lib/modules/selectedEntry.js

```diff
@@ -167,12 +167,36 @@ function _selectedThing() {
 	return selectedThing;
 }
 
-const listeners = new $.Callbacks();
+const listeners = { beforeScroll: [], beforePaint: [], idle: [] };
 
-export function addListener(callback) {
-	listeners.add(callback);
+export function addListener(callback, when = 'idle') {
+	listeners[when].push(callback);
 }
 
+const runCallbacks = (() => {
+	function runListeners(listeners, newSelected, oldSelected, options) {
+		for (const listener of listeners) listener(newSelected, oldSelected, options);
+	}
+
+	let idleCallback;
+	const requestIdle = window.requestIdleCallback ?
```

Comment:
Should probably move this to `utils/async.js` (maybe not export it for now, until we need something other than `idleThrottle` in module code)